crates/sncast/src/response/explorer_link.rs

```diff
@@ -22,27 +24,29 @@ pub enum ExplorerError {
     UnrecognizedNetwork,
 }
 
-pub fn print_block_explorer_link_if_allowed<T: OutputLink>(
+pub fn block_explorer_link_if_allowed<T: OutputLink>(
     result: &anyhow::Result<T>,
     output_format: OutputFormat,
     chain_id: Felt,
     show_links: bool,
     explorer: Option<Service>,
-) {
+) -> Option<String> {
     if !show_links {
-        return;
+        return None;
     }
     if output_format != OutputFormat::Human {
-        return;
+        return None;
     }
     let Ok(response) = result else {
-        return;
+        return None;
     };
     let Ok(network) = chain_id.try_into() else {
-        return;
+        return None;
     };
 
     if let Ok(provider) = explorer.unwrap_or_default().as_provider(network) {
-        response.print_links(provider);
+        return Some(response.print_links(provider));
```

Comment:
Maybe the method should be renamed. It doesn't seem like it is printing anything anymore